File: app/src/main/java/org/schabi/newpipe/streams/io/SharpOutputStream.java
package org.schabi.newpipe.streams.io;

import androidx.annotation.NonNull;

import java.io.IOException;
import java.io.OutputStream;

/**
 * Simply wraps a writable {@link SharpStream} allowing it to be used with built-in Java stuff that
 * supports {@link OutputStream}.
 */
public class SharpOutputStream extends OutputStream {
    private final SharpStream stream;

    public SharpOutputStream(final SharpStream stream) throws IOException {
        if (!stream.canWrite()) {
            throw new IOException("SharpStream is not writable");
        }
        this.stream = stream;
    }

    @Override
    public void write(final int b) throws IOException {
        stream.write((byte) b);
    }

    @Override
    public void write(@NonNull final byte[] b) throws IOException {
        stream.write(b);
    }

    @Override
    public void write(@NonNull final byte[] b, final int off, final int len) throws IOException {
        stream.write(b, off, len);
    }

    @Override
    public void flush() throws IOException {
        stream.flush();
    }

    @Override
    public void close() {
        stream.close();
    }
}

File: app/src/main/java/org/schabi/newpipe/streams/io/SharpOutputStream.kt
package org.schabi.newpipe.streams.io

import java.io.OutputStream
import okio.IOException

/**
 * Simply wraps a writable [SharpStream] allowing it to be used with built-in Java stuff that
 * supports [OutputStream].
 */
class SharpOutputStream
@Throws(IOException::class)
constructor(
    private val stream: SharpStream
) : OutputStream() {

    init {
        if (!stream.canWrite()) {
            throw IOException("SharpStream is not writable")
        }
    }

    @Throws(IOException::class)
    override fun write(b: Int) {
        stream.write(b.toByte())
    }

    @Throws(IOException::class)
    override fun write(b: ByteArray) {
        stream.write(b)
    }

    @Throws(IOException::class)
    override fun write(b: ByteArray, off: Int, len: Int) {
        stream.write(b, off, len)
    }

    @Throws(IOException::class)
    override fun flush() {
        stream.flush()
    }

    override fun close() {
        stream.close()
    }
}
